[package] qt/5.15.2: qt5 winextras needs to link dwmapi.lib

I am using qt 5.15.2 with the winextras option enabled

```
    def requirements(self):
        self.requires("qt/5.15.2")

    def configure(self):
        self.options["qt"].qtwinextras = True
```

In CMake I am doing a 
```
find_package(Qt5 COMPONENTS
  WinExtras
)

#...

 target_link_libraries (${PROJECT_NAME}
   Qt5::WinExtras
  )
```

However the build fails with a ton of linker errors:
```
error LNK2019: unresolved external symbol __imp_DwmGetColorizationColor referenced in function "class QColor __cdecl QtWin::colorizationColor(bool *)" (?colorizationColor@QtWin@@YA?AVQColor@@PEA_N@Z)
```

Adding `dwmapi.lib` for the linker resolves the problem.
I guess this needs to be modelled in the recipe.

--- a/recipes/qt/5.x.x/conanfile.py
+++ b/recipes/qt/5.x.x/conanfile.py
@@ -1077,7 +1077,6 @@ Examples = bin/datadir/examples""")
 
         if self.options.qtwinextras:
             _create_module("WinExtras")
-            self.cpp_info.components["qtWinExtras"].system_libs.append("dwmapi")
 
         if self.options.qtxmlpatterns:
              _create_module("XmlPatterns", ["Network"])
@@ -1090,7 +1089,10 @@ Examples = bin/datadir/examples""")
                 self.cpp_info.components["qtCore"].system_libs.append("userenv")  # qtcore requires "__imp_GetUserProfileDirectoryW " which is in "UserEnv.Lib" library
                 self.cpp_info.components["qtCore"].system_libs.append("ws2_32")  # qtcore requires "WSAStartup " which is in "Ws2_32.Lib" library
                 self.cpp_info.components["qtNetwork"].system_libs.append("dnsapi")  # qtnetwork from qtbase requires "DnsFree" which is in "Dnsapi.lib" library
-                self.cpp_info.components["qtNetwork"].system_libs.append("iphlpapi")
+                self.cpp_info.components["qtNetwork"].system_libs.append("iphlpapi")            
+                if self.options.qtwinextras:
+                    self.cpp_info.components["qtWinExtras"].system_libs.append("dwmapi")  # qtwinextras requires "DwmGetColorizationColor" which is in "dwmapi.lib" library
+
 
             if self.settings.os == "Macos":
                 self.cpp_info.components["qtCore"].frameworks.append("IOKit")     # qtcore requires "_IORegistryEntryCreateCFProperty", "_IOServiceGetMatchingService" and much more which are in "IOKit" framework

--- a/recipes/qt/5.x.x/conanfile.py
+++ b/recipes/qt/5.x.x/conanfile.py
@@ -1089,7 +1089,7 @@ Examples = bin/datadir/examples""")
                 self.cpp_info.components["qtCore"].system_libs.append("userenv")  # qtcore requires "__imp_GetUserProfileDirectoryW " which is in "UserEnv.Lib" library
                 self.cpp_info.components["qtCore"].system_libs.append("ws2_32")  # qtcore requires "WSAStartup " which is in "Ws2_32.Lib" library
                 self.cpp_info.components["qtNetwork"].system_libs.append("dnsapi")  # qtnetwork from qtbase requires "DnsFree" which is in "Dnsapi.lib" library
-                self.cpp_info.components["qtNetwork"].system_libs.append("iphlpapi")            
+                self.cpp_info.components["qtNetwork"].system_libs.append("iphlpapi")
                 if self.options.qtwinextras:
                     self.cpp_info.components["qtWinExtras"].system_libs.append("dwmapi")  # qtwinextras requires "DwmGetColorizationColor" which is in "dwmapi.lib" library
 

--- a/recipes/qt/5.x.x/conanfile.py
+++ b/recipes/qt/5.x.x/conanfile.py
@@ -1077,6 +1077,7 @@ Examples = bin/datadir/examples""")
 
         if self.options.qtwinextras:
             _create_module("WinExtras")
+            self.cpp_info.components["qtWinExtras"].system_libs.append("dwmapi")
 
         if self.options.qtxmlpatterns:
              _create_module("XmlPatterns", ["Network"])

